streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordQueue.java

```diff
@@ -104,10 +105,27 @@ public TopicPartition partition() {
      * @param rawRecords the raw records
      * @return the size of this queue
      */
-    int addRawRecords(final Iterable<ConsumerRecord<byte[], byte[]>> rawRecords) {
+    int addRawRecords(final Iterable<ConsumerRecord<byte[], byte[]>> rawRecords,
+                      final long consumerPosition) {
         for (final ConsumerRecord<byte[], byte[]> rawRecord : rawRecords) {
             fifoQueue.addLast(rawRecord);
         }
+        if (consumerPosition <= this.consumerPosition) {
+            throw new IllegalArgumentException(String.format(
+                "Consumer position must go forward; current %d, new %d",
+                this.consumerPosition,
+                consumerPosition));
+        }
+        if (!fifoQueue.isEmpty()) {
```

Comment:
This could happen if the buffer is empty and if the consumer only stepped over a commit marker passing in empty list of records.